src/memory_pool.c

```diff
@@ -261,7 +276,16 @@ static umf_result_t umfPoolCreateInternal(const umf_memory_pool_ops_t *ops,
         }
     }
 
+    if (ops->ext_post_initialize != NULL) {
+        ret = ops->ext_post_initialize(pool->provider, params, pool->pool_priv);
```

Comment:
document what happens if post initialize fails. Who is responsible for clearing things initialized in initialize bunction